src/components/configurator/settings/import-button.tsx

```diff
@@ -0,0 +1,230 @@
+/*
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+"use client"
+
+import { useImportConfig } from "@/api/use-import-config"
+import { Button } from "@/components/ui/button"
+import { useDevice } from "@/components/providers/device-provider"
+import { Loader2, Upload, X } from "lucide-react"
+import { useEffect, useRef, useState } from "react"
+import { CompatibilityDialog } from "./compatibility-dialog"
+import { useToast } from "@/components/ui/use-toast"
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from "@/components/ui/tooltip"
+import { Badge } from "@/components/ui/badge"
+
+export function ImportButton() {
+  const device = useDevice()
+  const {
+    importConfig,
+    isImporting,
+    showCompatibilityDialog,
+    currentConfigFile,
+    compatibilityIssues,
+    handleCompatibilityContinue,
+    handleCompatibilityCancel,
+    getCurrentFirmwareVersion
+  } = useImportConfig()
+  const fileInputRef = useRef<HTMLInputElement>(null)
+  const [currentFirmwareVersion, setCurrentFirmwareVersion] = useState<string>("Unknown")
+  const [isHovered, setIsHovered] = useState(false)
+  const [selectedFileName, setSelectedFileName] = useState<string | null>(null)
+  const { toast } = useToast()
+  const [importCancellable, setImportCancellable] = useState(false)
+  const [importToastId, setImportToastId] = useState<string | null>(null)
+
+  // Get current firmware version
+  useEffect(() => {
+    const fetchFirmwareVersion = async () => {
```

Comment:
Consider using Tanstack Query like other APIs in the `api/` directory e.g., `use-firmware-version.ts`. It has caching, query invalidation, etc.